cloudimg/aws.py

```diff
@@ -40,18 +57,22 @@ class AWSPublishingMetadata(PublishingMetadata):
                                            interfaces. By default this option
                                            is enabled.
         billing_products (list, optional): Billing product identifiers
+
+        boot_mode (str, optional): The boot mode for booting up the AMI on EC2.
     """
 
     def __init__(self, *args, **kwargs):
         self.ena_support = kwargs.pop('ena_support', True)
         self.sriov_net_support = kwargs.pop('sriov_net_support', 'simple')
         self.billing_products = kwargs.pop('billing_products', None)
+        bmode_str = kwargs.pop('boot_mode', None) or "not_set"
+        self.boot_mode = AWSBootMode[bmode_str]
 
         super(AWSPublishingMetadata, self).__init__(*args, **kwargs)
 
-        # Convert uefi_support bool into AWS specific values
-        self.boot_mode = "uefi-preferred" if self.uefi_support \
-            else "legacy-bios"
+        # Set the UEFI mode when the boolean is used instead of Enum.
+        if self.boot_mode == AWSBootMode.not_set and self.uefi_support is True:
+            self.boot_mode = AWSBootMode.uefi
```

Comment:
```
if self.boot_mode == AWSBootMode.not_set and self.uefi_support is not None:
     self.boot_mode = AWSBootMode.hybrid if self.uefi_support else AWSBootMode.bios
```
IIRC we had this in one of previous versions :-)

and it might be good idea to add unit test for **missing** `boot_mode` and `True/False` for `uefi_support`
Just to ensure we will not break pubtools-ami expectation :-)

Reply:
It's fixed now, PTAL again